### PR TITLE
[ENH] Implemented SimpleImputer

### LINKED ISSUE #2293: [ENH] Implement SimpleImputer class
### Describe the feature or idea you want to propose

It would be good to have imputers for  time series. sklearn has 
https://scikit-learn.org/1.5/api/sklearn.impute.html

SimpleImputer: 
KNNImputer: 
IterativeImputer

But these are not appropriate for time series, since they work across features (or time points for series) sklearn imputers work across time points not series.

](https://scikit-learn.org/1.5/modules/generated/sklearn.impute.IterativeImputer.html#sklearn.impute.IterativeImputer)

### Describe your proposed solution

file in transformations/collections called _impute.py with separate classes for different types of imputation. I dont mind copying sklearn structure, we should definitely have a version of SimpleImputer:

https://scikit-learn.org/1.5/modules/generated/sklearn.impute.SimpleImputer.htm

SimpleImputer: replace missing with a constant, default series mean. sklearn has a strategy argument. Note we operate along series not columns:

strategy: str or Callable, default=’mean’
The imputation strategy.
If “mean”, then replace missing values using the mean along each column. Can only be used with numeric data.
If “median”, then replace missing values using the median along each column. Can only be used with numeric data.
If “most_frequent”, then replace missing using the most frequent value along each column. Can be used with strings or numeric data. If there is more than one such value, only the smallest is returned.
If “constant”, then replace missing values with fill_value. Can be used with strings or numeric data.
If an instance of Callable, then replace missing values using the scalar statistic returned by running the callable over a dense 1d array containing non-missing values of each column.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by notaryanramani**
#### Reference Issues/PRs

Fixes #2293 

#### What does this implement/fix? Explain your changes.

Implements `SimpleImputer` class in `aeon.transformations.collections`, along with test cases.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
thanks for this, but I think the imputation should be for each case and channel independently not for all cases. Given it can be a list, and unequal length, its probably easiest just to iterate over series and do them like that

File: aeon/transformations/collection/_impute.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            missing values with on each 1D array containing all
+            non-missing values of each series.
+
+    fill_value : float, int, str or None, default=None
+        The value to replace missing values with. Only used when strategy is "constant".
+    """
+
+    _tags = {
+        "X_inner_type": ["np-list", "numpy3D"],
+        "fit_is_empty": False,
```
[the remark this answers is not in the document]
Comment (notaryanramani) [reply aeon/transformations/collection/_impute.py:41]: Yes, there is no fit method. Actually I just copied the tags from [here](https://github.com/aeon-toolkit/aeon/issues/2293#issuecomment-2471059166) and skipped to check it 😅.  Will fix that

**Comment by TonyBagnall** [comment]
thanks but this is still not logically correct, or rather not how I would like it. I want mean/median interpolation to happen independently on each channel and I dont think it is doing that at the moment. Logically
```python
n_channels = X[0].shape(0)
for i in range(len(X)):
    for j in range(n_channels):
           interpolate any missing in X[i][j]
```
Its obviously possible you can do it more efficiently without loops, but I think the outer loop is fine because it generalises the case for list and 3D array collections

**Comment by notaryanramani** [comment]
I think the parameter `axis=-1` takes care that we need the mean of every channel independently.

**Review by TonyBagnall — APPROVED** [review]
LGTM now, thanks

**Comment by TonyBagnall** [comment]
@all-contributors add @notaryanramani for code
